### DISCUSSION TITLE
Import other elevation data sources from OpenTopography

### DISCUSSION THREAD

**Discussion by klarrieu** [domlysz/BlenderGIS#682] [Ideas]
Global topographic datasets other than SRTM (Copernicus, ALOS, SRTM+, etc.) are now available via the OpenTopography API, and with greater global coverage than the SRTM dataset (e.g. high latitudes). Adding the ability to import from these other data sources could be useful.

**Comment by rhclayto** [comment]
I agree. There are a lot of data sources on OpenTopography. I am personally interested in the USGS 1/3 arc-second Digital Elevation Model. It is higher resolution by a lot than SRTM. In any case, having selectable data sources in this program would be extremely nice!

Edit: Okay, at https://opentopography.org/developers it looks like only the following sources are available via the REST API: 'Shuttle Radar Topography Mission [GL3 (Global 90m)](https://portal.opentopography.org/raster?opentopoID=OTSRTM.042013.4326.1), [GL1 (Global 30m)](https://portal.opentopography.org/raster?opentopoID=OTSRTM.082015.4326.1), [GL1 Ellipsoidal](https://portal.opentopography.org/raster?opentopoID=OTSRTM.082016.4326.1), [ALOS World 3D (Global 30m)](https://portal.opentopography.org/raster?opentopoID=OTALOS.112016.4326.2), [ALOS World 3D Ellipsoidal](https://portal.opentopography.org/raster?opentopoID=OTALOS.082017.4326.1), [Global Bathymetry and Topography at 15 Arc Sec: SRTM15+ V2.1](https://portal.opentopography.org/datasetMetadata?otCollectionID=OT.122019.4326.1), NASADEM (NASADEM Global DEM), COP30 (Copernicus Global DSM 30m) and COP90 (Copernicus Global DSM 90m) data.'

Pity. Still would be good to be able to select from amongst what is available.

**Reply by cjcrosby** [reply]
Stumbled on this ticket. FYI, last fall we launched a second API that retrieves USGS 3DEP raster (DEM) data - this includes 1 arc-second (~30m), 1/3 arc-second (~10m), and 1m spatial resolution data: https://opentopography.org/news/api-access-usgs-3dep-rasters-now-available

The API behaves very similarly to the Global Datasets API already in BlenderGIS. Existing OpenTopography API keys will work. API documentation: https://portal.opentopography.org/apidocs/#/Public/getUsgsDem

**Comment by domlysz** [comment]
You can already setup other server url for elevation data in the addon preferences
![Capture d’écran du 2022-09-07 10-10-05](https://user-images.githubusercontent.com/5639024/188895382-0f7719f0-3b8f-4dae-ae34-e9d8d8494048.png)

**Comment by rhclayto** [comment]
That's good to know. I wrote to Opentopography, & asked if they would open their other data via API. They said it's something they have been considering, & may be a feature we see in future.

-----

Are there any good tutorials showing how to match imported elevation data (DEM GeoTIFFs for example) with map base layers (Google, ESRI, etc.)? Especially in UTM instead of Web Mercator. What I'm hoping to be able to do is match the best elevation data (for my project, USGS 1/3 arc-second DEM) with the best map (for my project, ESRI), & have accurate distance measurements (UTM).
